src/main/java/com/avispl/symphony/dal/communicator/nec/multisync/NECMultisyncDevice.java

```diff
@@ -65,28 +113,38 @@ public void setHistoricalProperties(String historicalProperties) {
 
     /**
      * This method is recalled by Symphony to control specific property
-     * @param controllableProperty This is the property to be controled
+     *
+     * @param controllableProperty This is the property to be controlled
      */
     @Override
     public void controlProperty(ControllableProperty controllableProperty) throws Exception {
-        powerLock.lock();
+        reentrantLock.lock();
```

Comment:
please put this right before the try{} block - _potentially_ calls below this point may end up with an exception (NPEs for instance)